examples/dreamcast/filesystem/browse/browse.c

```diff
@@ -0,0 +1,379 @@
+    DIR *d;
+    struct dirent *entry;
+
+    /* Open the directory */
+    if(!(d = opendir(directory)));
+
+    /* Clear out all files */
+    memset(directory_contents, 0, sizeof(directory_contents[0])*100);
+
+    /* Read all the filenames in the directory */
+    while((entry = readdir(d)) && count < 100) {
+        directory_contents[count].is_dir = (entry->d_type == DT_DIR);
+        strncpy(directory_contents[count].filename, entry->d_name, strlen(entry->d_name));
+        count++;
+    }
+
+    /* Close directory */
+    closedir(d);
+
+    return count;
+}
+
+static void prompt_message(char *message, bool highlight_yes) {
+    int x = 20 + 24, y = 350;
+    int color = 1;
+
+    bfont_set_foreground_color(0xFFFFFFFF);
+    bfont_set_background_color(0x00000000);
+    bfont_draw_str(vram_s + y*640+x, 640, color, message);
+
+    if(highlight_yes) {
+        bfont_set_foreground_color(0x00000000);
+        bfont_set_background_color(0xFFFFFFFF);
+    }
+    else {
+        bfont_set_foreground_color(0xFFFFFFFF);
+        bfont_set_background_color(0x00000000);
+    }
+    y += 24;
+    bfont_draw_str(vram_s + y*640+x, 640, color, "YES");
+
+    if(highlight_yes) {
+        bfont_set_foreground_color(0xFFFFFFFF);
+        bfont_set_background_color(0x00000000);
+    }
+    else {
+        bfont_set_foreground_color(0x00000000);
+        bfont_set_background_color(0xFFFFFFFF);
+    }
+    y += 24;
+    bfont_draw_str(vram_s + y*640+x, 640, color, "NO");
+}
+
+static void draw_directory_selector(int index) {
+    int x = 24, y = 24 + (index  *24);
+    int color = 1;
+
+    bfont_set_foreground_color(0xFFFFFFFF);
+    bfont_set_background_color(0x00000000);
+    bfont_draw_str(vram_s + y*640+x, 640, color, ">");
+}
+
+static void draw_directory_contents(directory_file_t directory_contents[], int num) {
+    int x = 20 + 24, y = 24;
+    int color = 1;
+    int count = 0;
+    char str[80];
+
+    bfont_set_foreground_color(0xFFFFFFFF);
+    bfont_set_background_color(0x00000000);
+    for(int i=0;i<num;i++) {
+        if(directory_contents[i].is_dir) {
+             sprintf(str, "%-40s%s", directory_contents[i].filename, "< DIR >");
+             bfont_draw_str(vram_s + y*640+x, 640, color, str);
+        }
+        else {
+            bfont_draw_str(vram_s + y*640+x, 640, color, directory_contents[i].filename);
+        }
+        count++;
+        y += 24;
+
+        if(y >= (480 - 24))
+            break;
+    }
+}
+
+static cont_state_t *get_cont_state() {
+    maple_device_t *cont;
+    cont_state_t *state;
+
+    cont = maple_enum_type(0, MAPLE_FUNC_CONTROLLER);
+    if(cont) {
+        state = (cont_state_t*)maple_dev_status(cont);
+        return state;
+    }
+
+    return NULL;
+}
+
+static int button_pressed(unsigned int current_buttons, unsigned int changed_buttons, unsigned int button) {
+    if (changed_buttons & button) {
+        if (current_buttons & button)
+            return 1;
+    }
+
+    return 0;
+}
```

Comment:
```suggestion
static bool button_pressed(unsigned int current_buttons, unsigned int changed_buttons, unsigned int button) {
    if (changed_buttons & button) {
        if (current_buttons & button)
            return true;
    }

    return false;
}
```
or maybe a slightly less readable
```suggestion
static bool button_pressed(unsigned int current_buttons, unsigned int changed_buttons, unsigned int button) {
    return !!(changed_buttons & current_buttons & button);
}
```